src/Analyser/TypeSpecifier.php

```diff
@@ -482,6 +482,8 @@ public function specifyTypesInCondition(
 					$argType = $scope->getType($expr->right->getArgs()[0]->value);
 					if ($argType->isArray()->yes()) {
 						$result = $result->unionWith($this->create($expr->right->getArgs()[0]->value, new NonEmptyArrayType(), $context, false, $scope));
+					} else {
+						$result = $result->unionWith($this->create($expr->right->getArgs()[0]->value, new ConstantArrayType([], []), $context->negate(), false, $scope));
```

Comment:
same thing as a couple lines further down but for `is_array($a) && count($a) > 0 || null === $a`